Source/SvgDocument.cs

```diff
@@ -280,139 +280,177 @@ public static T Open<T>(Stream stream, Dictionary<string, string> entities) wher
-
-                foreach (var rule in sheet.StyleRules)
+                if (styles.Any())
                 {
-                    aggList = rule.Selector as AggregateSelectorList;
-                    if (aggList != null && aggList.Delimiter == ",")
-                    {
-                        selectors = aggList;
-                    }
-                    else
+                    var cssTotal = styles.Select((s) => s.Content).Aggregate((p, c) => p + Environment.NewLine + c);
+                    var cssParser = new Parser();
+                    var sheet = cssParser.Parse(cssTotal);
+                    AggregateSelectorList aggList;
+                    IEnumerable<BaseSelector> selectors;
+                    IEnumerable<SvgElement> elemsToStyle;
+
+                    foreach (var rule in sheet.StyleRules)
                     {
-                        selectors = Enumerable.Repeat(rule.Selector, 1);
-                    }
+                        aggList = rule.Selector as AggregateSelectorList;
+                        if (aggList != null && aggList.Delimiter == ",")
+                        {
+                            selectors = aggList;
+                        }
+                        else
+                        {
+                            selectors = Enumerable.Repeat(rule.Selector, 1);
+                        }
 
-                    foreach (var selector in selectors)
-                    {
-                        try
+                        foreach (var selector in selectors)
                         {
-                            elemsToStyle = svgDocument.QuerySelectorAll(rule.Selector.ToString(), elementFactory);
-                            foreach (var elem in elemsToStyle)
+                            try
                             {
-                                foreach (var decl in rule.Declarations)
+                                elemsToStyle = svgDocument.QuerySelectorAll(rule.Selector.ToString(), elementFactory);
+                                foreach (var elem in elemsToStyle)
                                 {
-                                    elem.AddStyle(decl.Name, decl.Term.ToString(), rule.Selector.GetSpecificity());
+                                    foreach (var decl in rule.Declarations)
+                                    {
+                                        elem.AddStyle(decl.Name, decl.Term.ToString(), rule.Selector.GetSpecificity());
+                                    }
                                 }
                             }
-                        }
-                        catch (Exception ex)
-                        {
-                            Trace.TraceWarning(ex.Message);
+                            catch (Exception ex)
+                            {
+                                Trace.TraceWarning(ex.Message);
+                                if(ExceptionCaughtIsGdiPlusRelated(ex)) { throw; } //Gdi+ errors should be rethrown

```

Comment:
This block is application of styles.
In my understanding, I think that GDI+ is irrelevant in this process.

Is this necessary ?